server/swap/swap.go

```diff
@@ -340,6 +340,12 @@ func (s *Swapper) deleteMatch(mt *matchTracker) {
 	mid := mt.ID()
 	delete(s.matches, mid)
 
+	// Unlock the maker and taker order coins. May be redundant if processBlock
+	// confirmed both swaps, but premature/quick counterparty actions that
+	// advance match status first prevent that.
+	s.unlockOrderCoins(mt.Maker)
+	s.unlockOrderCoins(mt.Taker)
```

Comment:
moved from `revoke` to `deleteMatch` so this happens in the successful completion path too (see `processRedeem` for the taker)